datasets/austin/_images/run_csv_transform_kub_311_service_requests/csv_transform.py

```diff
@@ -0,0 +1,206 @@
+
+
+import datetime
+import logging
+import os
+import pathlib
+
+import pandas as pd
+import requests
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+) -> None:
+
+    logging.info("Austin 311 Service Requests By Year process started")
+
+    logging.info("creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info(f"Downloading file {source_url}")
+    download_file(source_url, source_file)
+
+    logging.info(f"Opening file {source_file}")
+    df = pd.read_csv(str(source_file))
+
+    logging.info(f"Transforming.. {source_file}")
+    rename_headers(df)
+    convert_values(df)
+    delete_newlines_from_column(df, col_name="location")
+    filter_null_rows(df)
+
+    logging.info(f"Saving to output file.. {target_file}")
+    try:
+        save_to_new_file(df, file_path=str(target_file))
+    except Exception as e:
+        logging.error(f"Error saving output file: {e}.")
+    logging.info("..Done!")
+
+    logging.info(
+        f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path}"
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+
+def rename_headers(df: pd.DataFrame) -> None:
+    header_names = {
+        "Service Request (SR) Number": "unique_key",
+        "SR Type Code": "complaint_type",
+        "SR Description": "complaint_description",
+        "Owning Department": "owning_department",
+        "Method Received": "source",
+        "SR Status": "status",
+        "Status Change Date": "status_change_date",
+        "Created Date": "created_date",
+        "Last Update Date": "last_update_date",
+        "Close Date": "close_date",
+        "SR Location": "incident_address",
+        "Street Number": "street_number",
+        "Street Name": "street_name",
+        "City": "city",
+        "Zip Code": "incident_zip",
+        "County": "county",
+        "State Plane X Coordinate": "state_plane_x_coordinate",
+        "State Plane Y Coordinate": "state_plane_y_coordinate",
+        "Latitude Coordinate": "latitude",
+        "Longitude Coordinate": "longitude",
+        "(Latitude.Longitude)": "location",
+        "Council District": "council_district_code",
+        "Map Page": "map_page",
+        "Map Tile": "map_tile",
+    }
+
+    for old_name, new_name in header_names.items():
+        df.rename(old_name, new_name)
+
+
+def convert_dt_format(dt_str: str) -> str:
+    if str(dt_str).strip()[3:3] == "/":
+        return datetime.datetime.strptime(str(dt_str), "%m/%d/%Y %H:%M:%S %p").strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+    elif (
+        dt_str is None
+        or len(str(dt_str)) == 0
+        or str(dt_str).lower() == "nan"
+        or str(dt_str).lower() == "nat"
+    ):
+        return ""
+    else:
+        return str(dt_str)
```

Comment:
Same as above, stringifying a value that's already expected to be a string seems redundant.